### ISSUE TITLE
Parse `+json` content types suffix for web action bodies

### ISSUE THREAD

**Issue by akrabat** [openwhisk/openwhisk#2147]
It would be nice if I could post with a content type of `vnd.akrabat.com/api+json` and have the web action automatically decode the JSON encoded body of my HTTP message.

The +json suffix if specified in [RFC 6839](https://tools.ietf.org/html/rfc6839#section-3.1).

**Comment by rabbah** [comment]
This is for `.http` web actions, where the controller passes the body intact to the action as __ow_body. It would recover the behavior we provide for `.json` web actions for example where the body is passes as a top level JSON object to the action as an argument.

@akrabat did I understand correctly?

**Comment by akrabat** [comment]
Yes. For `.http` web actions, a `Content-Type` of `application/json` and `xxx/yyy+json` should behave the same.

**Comment by duglin** [comment]
I just ran into this today (I think its either the same, or related, issue).  My function is in a docker container and the __ow_body is sometimes base64 encoded and sometimes not based on these two content-type values:
`application/cloudevents+json`
`application/cloudevents+json; charset=UTF-8`
The UTF one is given to me as raw json, while the other one is base64 encoded. IMO both should be raw json. 

Additionally, it would be really nice if there was a flag (perhaps as a header) given to the function so it knows whether the body is base64 encoded or not. While I know it might be possible to look at the content-type and figure it out, you're asking the function to mimic the logic in OW to make that determination and that's not only error prone but could be a lot of work in the case where the function may not actually know about all of the various content-types and its acting more like a pass-thru. It seems like it would be really nice (from a UX) perspective if there was some flag someplace that the function could look at and know (w/o a doubt and w/o a lot of code) whether it needs to decode the body or not.

/cc @csantanapr

**Comment by csantanapr** [comment]
@rabbah what are the next steps to fix this in your opinion?

**Comment by rabbah** [comment]
On the utf8 non handling - that should be fixed; there is already an isJsonFamily method which is why one of them works but not the other. 

We can set a header or introduce another property to indicate b64 encoding. There is another related issue which is that extensions are not visible to the functions. If we do add a property we should address both at the same time.

**Comment by duglin** [comment]
@markusthoemmes that will definitely address my immediate problem so thanks!!

I do still think having a flag someplace that the function can check to know when to decode the __ow_body would be useful as well.  (yes I'm demanding 😄  )

**Comment by csantanapr** [comment]
Demand with PRs :-)

**Comment by rabbah** [comment]
I'm not sure if this is still an issue or not. Is there a test case?
